src/ai/backend/manager/models/blue_green_strategies.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+import logging
+
+import sqlalchemy as sa
+from sqlalchemy.orm import foreign, relationship
+
+from ai.backend.logging import BraceStyleAdapter
+
+from .base import (
+    Base,
+    IDColumn,
+)
+
+log = BraceStyleAdapter(logging.getLogger(__spec__.name))
+
+__all__ = ("BlueGreenStrategyRow",)
+
+
+def _get_deployment_to_bgs_join_cond():
+    from .model_deployment import ModelDeploymentRow
+
+    return ModelDeploymentRow.deployment_strategy_id == foreign(BlueGreenStrategyRow.id)
+
+
+class BlueGreenStrategyRow(Base):
+    __tablename__ = "blue_green_strategies"
+
+    id = IDColumn("id")
+
+    name = sa.Column("name", sa.String, nullable=True)
+
+    pre_switch_wait_time = sa.Column("pre_switch_wait_time", sa.Integer, default=60, nullable=False)
+    switch_timeout = sa.Column("switch_timeout", sa.Integer, default=30, nullable=False)
+    post_switch_wait_time = sa.Column(
+        "post_switch_wait_time", sa.Integer, default=60, nullable=False
+    )
```

Comment:
You need to receive the revision information corresponding to blue and green (blue is not nullable, and green is nullable).

Reply:
```
deployment_id
prev_revision_id
next_revision_id
strategy (enum) : [ROLLING, BLUE_GREEN]
strategy_id : Optional : ...
status : [PENDING, IN_PROGRESS, DONE, FAILED]
```